File: app/src/main/java/com/erotc/learning/activity/RecentResultActivity.java
package com.erotc.learning.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;

import com.erotc.learning.R;
import com.erotc.learning.dao.DictionaryEntry;
import com.erotc.learning.fragment.DetailFragment;
import com.erotc.learning.repository.DictionaryRepository;

import java.sql.SQLException;
import java.util.List;

public class RecentResultActivity extends AppCompatActivity {
    private static final String LOG_TAG = RecentResultActivity.class.getSimpleName();

    private ViewPager mPager;
    private PagerAdapter mPagerAdapter;

    private List<DictionaryEntry> mRecentResults;
    private DictionaryRepository mDictionaryRepository;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recent_result);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        mDictionaryRepository = DictionaryRepository.getInstance(this);

        try {
            mRecentResults = mDictionaryRepository.getRecentSearchResults(true);
        } catch (SQLException e) {
            e.printStackTrace();
            Log.e(LOG_TAG, Log.getStackTraceString(e));
        }

        mPager = findViewById(R.id.pager);
        mPagerAdapter = new ScreenSlidePagerAdapter(getSupportFragmentManager());
        mPager.setAdapter(mPagerAdapter);

        mPager.setCurrentItem((mRecentResults.size() - 1));
    }

    private class ScreenSlidePagerAdapter extends FragmentStatePagerAdapter {
        public ScreenSlidePagerAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            DictionaryEntry entry = mRecentResults.get(position);

            return DetailFragment.newInstance(entry.getId());
        }

        @Override
        public int getCount() {
            return mRecentResults.size();
        }

        @Nullable
        @Override
        public CharSequence getPageTitle(int position) {
            DictionaryEntry entry = mRecentResults.get(position);
            return entry.getTagalog();
        }
    }
}

File: app/src/main/java/com/erotc/learning/activity/RecentResultActivity.kt
package com.erotc.learning.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.PagerAdapter
import com.erotc.learning.R
import com.erotc.learning.data.DictionaryEntry
import com.erotc.learning.fragment.DetailFragment
import com.erotc.learning.repository.DictionaryRepository
import kotlinx.android.synthetic.main.activity_recent_result.*
import kotlinx.android.synthetic.main.content_recent_result.*

class RecentResultActivity : AppCompatActivity() {
    private var pagerAdapter: PagerAdapter? = null
    private var recentResults: List<DictionaryEntry>? = null

    private lateinit var dictionaryRepository: DictionaryRepository

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recent_result)
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        dictionaryRepository = DictionaryRepository.getInstance(this)
        recentResults = dictionaryRepository.getRecentSearchResults(true)

        pagerAdapter = ScreenSlidePagerAdapter(supportFragmentManager)
        
        pager.adapter = pagerAdapter
        pager.currentItem = recentResults?.size ?: 0 - 1
    }

    private inner class ScreenSlidePagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
        override fun getItem(position: Int): Fragment {
            val entry = recentResults?.get(position)
            return DetailFragment.newInstance(entry?.id ?: 0)
        }

        override fun getCount() = recentResults?.size ?: 0

        override fun getPageTitle(position: Int): CharSequence? {
            val entry = recentResults?.get(position)
            return entry?.tagalog
        }
    }

    companion object {
        private val LOG_TAG = RecentResultActivity::class.java.simpleName
    }
}